climada/engine/test/test_supplychain.py

```diff
@@ -0,0 +1,180 @@
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Test Supplychain class.
+"""
+
+import unittest
+import numpy as np
+
+from climada import CONFIG
+from climada.entity.exposures.base import Exposures
+from climada.entity import ImpactFuncSet, IFTropCyclone
+from climada.hazard.base import Hazard
+from climada.engine.supplychain import SupplyChain
+from climada.util.constants import EXP_DEMO_H5
+
+HAZ_TEST_MAT = CONFIG.hazard.test_data.dir().joinpath('atl_prob_no_name.mat')
+DIR_TEST_DATA = CONFIG.engine.test_data.dir()
+
+class TestSupplyChain(unittest.TestCase):
+    """Testing the SupplyChain class."""
+    def test_read_wiot(self):
+        """Test reading of wiod table."""
+        sup = SupplyChain()
+        sup.read_wiod(year='test', file_path=DIR_TEST_DATA)
+
+        self.assertAlmostEqual(sup.mriot_data[0, 0], 12924.1797, places=3)
+        self.assertAlmostEqual(sup.mriot_data[0, -1], 0, places=3)
+        self.assertAlmostEqual(sup.mriot_data[-1, 0], 0, places=3)
+        self.assertAlmostEqual(sup.mriot_data[-1, -1], 22.222, places=3)
+        
+        self.assertAlmostEqual(sup.mriot_data[0, 0], 
+                               sup.mriot_data[sup.cntry_pos[list(sup.cntry_pos)[0]][0], 
+                                              sup.cntry_pos[list(sup.cntry_pos)[0]][0]], 
+                               places=3)
+        self.assertAlmostEqual(sup.mriot_data[-1, -1], 
+                               sup.mriot_data[sup.cntry_pos[list(sup.cntry_pos)[-1]][-1], 
+                                              sup.cntry_pos[list(sup.cntry_pos)[-1]][-1]], 
+                               places=3)        
+        self.assertEqual(np.shape(sup.mriot_data), (112, 112))
+        self.assertAlmostEqual(sup.total_prod.sum(), 3533367.89439, places=3)
+        self.assertEqual(len(sup.countries_iso3), len(sup.countries))
+
+    def test_calc_impact(self):
+        """Test running direct and indirect impact calculations."""
+        
+        sup = SupplyChain()
+        sup.read_wiod(year='test', file_path=DIR_TEST_DATA)
+
+        # Tropical cyclone over Florida and Caribbean
+        hazard = Hazard('TC')
+        hazard.read_mat(HAZ_TEST_MAT)
+
+        # Read demo entity values
+        # Set the entity default file to the demo one
+        exp = Exposures()
+        exp.read_hdf5(EXP_DEMO_H5)
+        exp.check()
+        exp.region_id = 840 #assign right id for USA
+        exp.assign_centroids(hazard)
+
+        impf_tc= IFTropCyclone()
+        impf_tc.set_emanuel_usa()
+        impf_set = ImpactFuncSet()
+        impf_set.append(impf_tc)
+        impf_set.check()
+        
+        # Test direct impacts
+        sup.calc_sector_direct_impact(hazard, exp, impf_set,
```

Comment:
PLease add a test with `set_subsec = [...]`

Reply:
done in https://github.com/CLIMADA-project/climada_python/pull/180/commits/ed468290bf322d343c230549529e51c09832e9d0